python_modules/dagster-externals/dagster_externals/_io/default.py

```diff
@@ -1,57 +1,65 @@
+        self._validate_params(params)
+        self._path = params["path"]
+        yield
+
+    @property
+    def path(self) -> str:
+        assert self._path is not None
+        return self._path
 
-    def _get_path_from_env(self) -> str:
+    def _validate_params(self, params: ExternalExecutionParams) -> None:
         try:
-            context_injector_params = param_from_env_var("context")
-            assert isinstance(context_injector_params, dict)
-            assert isinstance(context_injector_params.get("path"), str)
-            return context_injector_params["path"]
+            assert isinstance(params.get("path"), str)
         except AssertionError:
+            param = "contxt" if isinstance(self, ExternalExecutionContextLoader) else "messages"
             raise DagsterExternalsError(
                 f"`{self.__class__.__name__}` requires a `path` key in the"
-                f" {DAGSTER_EXTERNALS_ENV_KEYS['context_injector']} environment variable be a JSON"
+                f" {DAGSTER_EXTERNALS_ENV_KEYS[param]} environment variable be a JSON"
                 " object with a string `path` property."
             )
 
 
-class ExternalExecutionFileMessageWriter(ExternalExecutionMessageWriter):
+class ExternalExecutionFileContextLoader(PathMixin, ExternalExecutionContextLoader):
+    def __init__(self):
+        self._path = None
+
+    def load_context(self) -> ExternalExecutionContextData:
+        with open(self.path, "r") as f:
+            return json.load(f)
+
+
+class ExternalExecutionFileMessageWriter(PathMixin, ExternalExecutionMessageWriter):
     def __init__(self):
         self._path = None
 
     def write_message(self, message: ExternalExecutionMessage) -> None:
         with open(self.path, "a") as f:
             f.write(json.dumps(message) + "\n")
 
-    @property
-    def path(self) -> str:
-        if self._path is None:
-            self._path = self._get_path_from_env()
-        return self._path
 
-    def _get_path_from_env(self) -> str:
-        try:
-            context_injector_params = param_from_env_var("messages")
-            assert isinstance(context_injector_params, dict)
-            assert isinstance(context_injector_params.get("path"), str)
-            return context_injector_params["path"]
-        except AssertionError:
-            raise DagsterExternalsError(
-                f"`{self.__class__.__name__}` requires a `path` key in the"
-                f" {DAGSTER_EXTERNALS_ENV_KEYS['message_reader']} environment variable be a JSON"
-                " object with a string `path` property."
-            )
+class ExternalExecutionEnvVarParamLoader(ExternalExecutionParamLoader):
+    def load_params(self) -> Tuple[Mapping[str, Any], Mapping[str, Any]]:
+        context_injector_params = param_from_env_var("context")
+        message_reader_params = param_from_env_var("messages")
+        return context_injector_params, message_reader_params
```

Comment:
since we have a class, seems like we could have separate methods for each parameter bundle, avoiding having to unpack tuple. Unnecessarily adds the mental overhead of having to remember/ensure you are not swapping the param bundles unnecessarily